packages/jellyfish-transaction/src/buffer/buffer_composer.ts

```diff
@@ -209,6 +209,32 @@ export abstract class ComposableBuffer<T> implements BufferComposer {
     }
   }
 
+  /**
+   *  HEX String without length specified, encoded into Buffer as the same order of the Hex String.
+   *  In short this read a hex and push it into the Buffer. It will not re-order the endian.
+   * `toBuffer` defaults to '' and writes 00 to buffer
+   *
+   * @param getter to read HEX String. Writes its length then write the HEX string. Defaults to empty string.
+   * @param setter to read ordered Buffer and set as the same ordered HEX String
+   */
+  static optionalVarUIntHex (getter: () => string | undefined, setter: (data: string | undefined) => void): BufferComposer {
+    return {
+      fromBuffer: (buffer: SmartBuffer): void => {
+        const length = readVarUInt(buffer)
+        if (length > 0) {
+          const buff = Buffer.from(buffer.readBuffer(length))
+          setter(buff.toString('hex'))
+        }
+      },
+      toBuffer: (buffer: SmartBuffer): void => {
+        const hex = getter() ?? ''
+        const buff: Buffer = Buffer.from(hex, 'hex')
+        writeVarUInt(buff.length, buffer)
+        buffer.writeBuffer(buff)
+      }
+    }
+  }
```

Comment:
The suggested change is more deterministic, best not to have multiple condition writing the same hex output.
Making `undefined` represent `0x00` and nothing else for a more bi-directional and consistent parsing.

```suggestion
  /**
   * HEX String with length specified, encoded into Buffer as the same order of the Hex String.
   * In short this read a hex and push it into the Buffer. It will not re-order the endian.
   *
   * When hex is `undefined` it will write `0x00` signifying zero byte length.
   *
   * @param getter to read HEX String. Writes its length then write the HEX string. Defaults to empty string.
   * @param setter to read ordered Buffer and set as the same ordered HEX String
   */
  static varUIntHex (getter: () => string | undefined, setter: (data: string | undefined) => void): BufferComposer {
    return {
      fromBuffer: (buffer: SmartBuffer): void => {
        const length = readVarUInt(buffer)
        if (length > 0) {
          const buff = Buffer.from(buffer.readBuffer(length))
          setter(buff.toString('hex'))
        }
      },
      toBuffer: (buffer: SmartBuffer): void => {
        const hex = getter()
        if (hex !== undefined ) {
          if (hex === '') { 
            throw new Error('ComposableBuffer.varUIntHex.toBuffer attempting to write empty buffer')
          }
          const buff: Buffer = Buffer.from(hex, 'hex')
          writeVarUInt(buff.length, buffer)
          buffer.writeBuffer(buff)
        } else {
          writeVarUInt(0x00, buffer)
        }
      }
    }
  }
```

Reply:
Okay let me change it.
What about the naming though ? It does not follow the same logic as `varUIntUtf8BE`, `varUIntUtf8LE` and `varUIntArray`

Reply:
Maybe `varUIntOptionalHex` moving the optional back since `varUInt` is always present but if 0 it gives an `OptionalHex`